lambdas/process_webhooks/process_webhooks.py

```diff
@@ -53,7 +63,257 @@ def _get_target_url(headers):
+            # PR was either "opened" or "synchronized" which is
+            # when a new commit is pushed to the PR
+            repository = payload['pull_request']['base']['repo']['name']
+            ref = payload['pull_request']['base']['ref']
+            sha = payload['pull_request']['head']['sha']
+        else:
+            ignore = True
+    else:
+        # Unsupported event type, return None for both values
+        ignore = True
+
+    # Find the target based on the base_ref
+    if not ignore:
+        if ref == "refs/heads/master":
+            target = "master"
+        elif ref in RELEASE_BRANCHES:
+            # find the target from constants.py
+            target = RELEASE_BRANCHES[ref]
+        else:
+            # no jobs are expected in this case
+            ignore = True
+
+    # If we are ignoring this hook, return None values,
+    # otherwise, return sha, jobs_list
+    if ignore:
+        # We don't care about these so assign None, None
+        return (None, None, None)
+    else:
+        # Find the jobs list for this hook
+        jobs_list = _get_jobs_list(repository, target)
+
+    return sha, jobs_list, target
+
+
+def _parse_executable_for_builds(
+    executable, build_status, event_type, target, hook_sha
+):
+    """
+    Parse executable to find the sha and job name of
+    queued/running builds.
+    Return list of jobs with the sha that triggered
+    them
+    """
+    builds = []
+    if event_type == "pull_request":
+        # All PR jobs show the sha that triggered them inside its
+        # parameters.
+        for action in executable['actions']:
+            if 'parameters' in action:
+                for param in action['parameters']:
+                    if (param['name'] == 'sha1' or
+                            param['name'] == 'ghprbActualCommit'):
+                        sha = param['value']
+                        if build_status == 'queued':
+                            job_name = executable['task']['name']
+                        elif build_status == 'running':
+                            url = executable['url']
+                            m = re.search(
+                                r'/job/([^/]+)/.*',
+                                urlparse(url).path
+                            )
+                            job_name = m.group(1)
+                        if sha == hook_sha:
+                            builds.append({
+                                'job_name': job_name,
+                                'sha': sha
+                            })
+
+    return builds
+
+
+def _get_queued_builds(
+    jenkins_url, jenkins_username, jenkins_token, event_type, target, sha
+):
+    """
+    Find all builds currently in the queue
+    """
+    build_status = 'queued'
+    builds = []
+
+    # Use Jenkins REST API to get info on the queue
+    # set a timeout for the request to avoid timing out of lambda
+    url = '%s/queue/api/json?depth=0' % (jenkins_url)
```

Comment:
nit(s): in a few places there are mixed string manipulators, i.e. old C style, concatenation and format. I think we should standardize on one, preferably 'format'